File: ss-won/programmers/43165.cpp
#include <string>
#include <vector>
using namespace std;

int search(vector<int> arr, int s, int curr, int target)
{
    int res = 0;
    if (s == arr.size())
    {
        if (curr == target)
            return 1;
        return 0;
    }
    res += search(arr, s + 1, curr + arr[s], target);
    res += search(arr, s + 1, curr - arr[s], target);
    return res;
}

int solution(vector<int> numbers, int target)
{
    int answer = 0;
    answer = search(numbers, 0, 0, target);
    return answer;
}

File: ss-won/programmers/43165.js
// 코딩테스트 고득점 Kit(DFS/BFS) - 타겟 넘버(43165)
function solution(numbers, target) {
  let answer = 0;
  answer = getmethods(numbers, 0, target);
  return answer;
}

function getmethods(arr, curr, target) {
  let res;
  if (arr.length === 0) return curr === target ? 1 : 0;
  const narr = arr.filter((v, i) => i != 0);
  return (res =
    getmethods(narr, curr + arr[0], target) +
    getmethods(narr, curr - arr[0], target));
}
